From 21f393313768b3a343d51e5c37accff79424a6a7 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Thu, 8 Aug 2019 22:15:50 +0200
Subject: [PATCH 1/2] Revert "bfq: Fix the missing barrier in
 __bfq_entity_update_weight_prio"

This reverts commit 5f3c6a8e4d3ae664216f5ebd0b73052d3649be7b.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-wf2q.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/block/bfq-wf2q.c b/block/bfq-wf2q.c
index 919374829b02..c789b03f8955 100644
--- a/block/bfq-wf2q.c
+++ b/block/bfq-wf2q.c
@@ -861,9 +861,7 @@ __bfq_entity_update_weight_prio(struct bfq_service_tree *old_st,
 
 		BFQ_BUG_ON(entity->tree && update_class_too);
 		BFQ_BUG_ON(old_st->wsum < entity->weight);
-		/* Matches the smp_wmb() in bfq_group_set_weight. */
-                smp_rmb();
-                old_st->wsum -= entity->weight;
+		old_st->wsum -= entity->weight;
 
 		if (entity->new_weight != entity->orig_weight) {
 			if (entity->new_weight < BFQ_MIN_WEIGHT ||
-- 
2.23.0.rc1

